File: Java/JsonBench.java
import java.util.HashMap;
import java.util.Map;

public class JsonBench {
    static final int ITERATIONS = 5000;
    
    public static void main(String[] args) {
        System.out.println("Java JSON Benchmark (" + ITERATIONS + " iterations)");
        
        String testJson = "{\"name\":\"test\",\"age\":42,\"active\":true,\"scores\":[1,2,3,4,5,6,7,8,9,10]}";
        
        long t0 = System.nanoTime();
        for (int i = 0; i < ITERATIONS; i++) {
            String[] parts = testJson.replace("{", "").replace("}", "").split(",");
        }
        long t1 = System.nanoTime();
        double ms = (t1 - t0) / 1e6;
        System.out.printf("Parse: %.2f ms (%.0f ops/sec)%n", ms, ITERATIONS / (ms / 1000));
        
        StringBuilder sb = new StringBuilder();
        t0 = System.nanoTime();
        for (int i = 0; i < ITERATIONS; i++) {
            sb.setLength(0);
            sb.append("{\"id\":").append(i)
              .append(",\"name\":\"user\"")
              .append(",\"value\":").append(i / 10.0)
              .append(",\"items\":[1,2,3,4,5,6,7,8,9,10]}");
            String result = sb.toString();
        }
        t1 = System.nanoTime();
        ms = (t1 - t0) / 1e6;
        System.out.printf("Serialize: %.2f ms (%.0f ops/sec)%n", ms, ITERATIONS / (ms / 1000));
        
        StringBuilder largeStr = new StringBuilder("{\"data\": \"");
        for (int i = 0; i < 1000000; i++) {
            largeStr.append("x");
        }
        largeStr.append("\"}");
        String large = largeStr.toString();
        t0 = System.nanoTime();
        for (int i = 0; i < ITERATIONS; i++) {
            int pos = large.indexOf("{\"");
        }
        t1 = System.nanoTime();
        ms = (t1 - t0) / 1e6;
        System.out.printf("Search: %.2f ms (%.0f ops/sec)%n", ms, ITERATIONS / (ms / 1000));
        
        t0 = System.nanoTime();
        for (int i = 0; i < ITERATIONS; i++) {
            int count = 0;
            int pos = 0;
            while ((pos = large.indexOf("\"\":", pos)) != -1) {
                count++;
                pos++;
            }
        }
        t1 = System.nanoTime();
        ms = (t1 - t0) / 1e6;
        System.out.printf("Count fields: %.2f ms (%.0f ops/sec)%n", ms, ITERATIONS / (ms / 1000));
    }
}

File: Python/json_bench.py
#!/usr/bin/env python3
import json
import time

ITERATIONS = 5000


def main():
    print(f"Python JSON Benchmark ({ITERATIONS} iterations)")

    test_data = {
        "name": "test",
        "age": 42,
        "active": True,
        "scores": [1, 2, 3, 4, 5, 6, 7, 8, 9, 10],
        "address": {"street": "Main", "city": "NYC"},
        "tags": ["a", "b", "c", "d", "e"],
    }
    test_json = json.dumps(test_data)
    large_str = '{"data": "' + "x" * 1000000 + '"}'

    t0 = time.perf_counter()
    for _ in range(ITERATIONS):
        data = json.loads(test_json)
    t1 = time.perf_counter()
    ms = (t1 - t0) * 1000
    print(f"Parse: {ms:.2f} ms ({ITERATIONS / (ms / 1000):.0f} ops/sec)")

    t0 = time.perf_counter()
    for i in range(ITERATIONS):
        result = json.dumps(
            {"id": i, "name": "user", "value": i / 10.0, "items": list(range(10))}
        )
    t1 = time.perf_counter()
    ms = (t1 - t0) * 1000
    print(f"Serialize: {ms:.2f} ms ({ITERATIONS / (ms / 1000):.0f} ops/sec)")

    t0 = time.perf_counter()
    for _ in range(ITERATIONS):
        pos = large_str.find('{"')
    t1 = time.perf_counter()
    ms = (t1 - t0) * 1000
    print(f"Search: {ms:.2f} ms ({ITERATIONS / (ms / 1000):.0f} ops/sec)")

    t0 = time.perf_counter()
    for _ in range(ITERATIONS):
        count = large_str.count('":')
    t1 = time.perf_counter()
    ms = (t1 - t0) * 1000
    print(f"Count fields: {ms:.2f} ms ({ITERATIONS / (ms / 1000):.0f} ops/sec)")


if __name__ == "__main__":
    main()
